client/src/hooks/helpers/useQuests.tsx

```diff
@@ -0,0 +1,485 @@
+  [
+    QuestName.BuildFarm,
+    [
+      { id: QuestType.CommonResources, title: "Common Resources" },
+      { id: QuestType.UncommonResources, title: "Uncommon Resources" },
+      { id: QuestType.RareResources, title: "Rare Resources" },
+      { id: QuestType.UniqueResources, title: "Unique Resources" },
+      { id: QuestType.LegendaryResources, title: "Legendary Resources" },
+      { id: QuestType.MythicResources, title: "Mythic Resources" },
+    ],
+  ],
+  [QuestName.BuildResource, [{ id: QuestType.Trade, title: "Donkeys and Lords" }]],
+  [QuestName.CreateTrade, [{ id: QuestType.Military, title: "Claim Starting Army" }]],
+  [QuestName.CreateArmy, [{ id: QuestType.Earthenshard, title: "Claim Earthen Shard" }]],
+  [QuestName.Travel, [{ id: QuestType.Travel, title: "Travel" }]],
+  [QuestName.BuildWorkersHut, [{ id: QuestType.Population, title: "Population" }]],
+  [QuestName.Market, [{ id: QuestType.Market, title: "Market" }]],
+  [QuestName.Pillage, [{ id: QuestType.Pillage, title: "Pillage" }]],
+  [QuestName.Mine, [{ id: QuestType.Mine, title: "Mine" }]],
+  [QuestName.Contribution, [{ id: QuestType.Contribution, title: "Contribution" }]],
+  [QuestName.Hyperstructure, [{ id: QuestType.Hyperstructure, title: "Hyperstructure" }]],
+]);
+
+export interface Quest {
+  name: QuestName;
+  description: string;
+  steps: string[];
+  prizes: Prize[];
+  depth: number;
+  status: QuestStatus;
+}
+
+export interface Prize {
+  id: QuestType;
+  title: string;
+}
+
+export enum QuestStatus {
+  InProgress,
+  Completed,
+  Claimed,
+}
+
+export const useQuests = () => {
+  const {
+    buildingQuantities,
+    entityArmies,
+    orders,
+    pillageHistoryLength,
+    hasTroops,
+    hasTraveled,
+    fragmentMines,
+    hyperstructures,
+    hyperstructureContributions,
+  } = useQuestDependencies();
+
+  const { questClaimStatus } = useQuestClaimStatus();
+
+  const settleQuest = useMemo(() => {
+    return {
+      name: QuestName.Settle,
+      description: "A gift of food from the gods",
+      steps: ["Settle your first Realm"],
+      prizes: questPrize.get(QuestName.Settle) || [],
+      depth: 0,
+      status: questClaimStatus[QuestName.Settle] ? QuestStatus.Claimed : QuestStatus.Completed,
+    };
+  }, [questClaimStatus[QuestName.Settle]]);
+
+  const buildFarmQuest = useMemo(
+    () => ({
+      name: QuestName.BuildFarm,
+      description: "Wheat is the lifeblood of your people. Go to the construction menu and build a farm",
+      status:
+        buildingQuantities.farms > 0
+          ? questClaimStatus[QuestName.BuildFarm]
+            ? QuestStatus.Claimed
+            : QuestStatus.Completed
+          : QuestStatus.InProgress,
+      steps: [
+        "Navigate to the construction menu",
+        "Select the 'Farm' card",
+        "Left click on a hex to build it, or right click to cancel",
+      ],
+      prizes: questPrize.get(QuestName.BuildFarm) || [],
+      depth: 1,
+    }),
```

Comment:
except for the status, can we merge all of this information with the QuestPrize object and put this in another file? It would make this one way more readable imo